polyglot/piranha/demo/match_only/ts/configurations/rules.toml

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2022 Uber Technologies, Inc.
+#
+# <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License at
+# <p>http://www.apache.org/licenses/LICENSE-2.0
+#
+# <p>Unless required by applicable law or agreed to in writing, software distributed under the
+# License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+[[rules]]
+name = "find_fors"
+query = """
+(
+    (for_statement) @for_stmt
+)
+"""
+
+[[rules]]
+name = "find_fors_within_functions"
+query = """
+(
+    (for_statement) @for_stmt
+)
+"""
+[[rules.constraints]]
+matcher = "(function_declaration) @fs"
+queries=[]
+
+[[rules]]
+name = "find_fors_within_functions_without_whiles"
```

Comment:
Could we rename the rule name to `find_fors_within_functions_not_within_whiles`?
it makes the name more consistent and clear IMO.
This will lead to other renamings (e.g., tests, `rules.toml`...)

Reply:
Sure, will do that soon.